DigitalOffice/Tests/ProjectServiceTests/Repositories/EditProjectByIdRepositoryTests.cs

```diff
@@ -0,0 +1,139 @@
+        private ProjectServiceDbContext dbContext;
+        private DbContextOptions<ProjectServiceDbContext> dbOptionsProjectService;
+
+        [SetUp]
+        public void SetUp()
+        {
+            dbOptionsProjectService = new DbContextOptionsBuilder<ProjectServiceDbContext>()
+                .UseInMemoryDatabase(databaseName: "ProjectServiceTest")
+                .Options;
+
+            dbContext = new ProjectServiceDbContext(dbOptionsProjectService);
+            repository = new ProjectRepository(dbContext);
+
+            departmentId = Guid.NewGuid();
+
+            existingProject = new DbProject
+            {
+                Name = "A test name",
+                DepartmentId = departmentId,
+                Description = "Description",
+                IsActive = true,
+                Deferred = false
+            };
+
+            editProject = new DbProject
+            {
+                Name = "Is different",
+                DepartmentId = Guid.NewGuid(),
+                Description = "Is different too",
+                IsActive = false,
+                Deferred = false
+            };
+        }
+
+        [Test]
+        public void SuccessfulEditProjectTest()
+        {
+            existingProject.Id = Guid.NewGuid();
+            editProject.Id = existingProject.Id;
+
+            dbContext.Add(existingProject);
+            dbContext.SaveChanges();
+            dbContext.Entry(existingProject).State = EntityState.Detached;
+            dbContext.SaveChanges();
+
+            var dbProject = dbContext.Projects
+                .AsNoTracking()
+                .SingleOrDefault(p => p.Id == existingProject.Id);
+            Assert.IsNotNull(dbProject);
+
+            repository.EditProjectById(editProject.Id, editProject);
+            dbContext.Entry(editProject).State = EntityState.Detached;
+            dbContext.SaveChanges();
+
+            var editedProject = dbContext.Projects
+                .AsNoTracking()
+                .SingleOrDefault(p => p.Id == editProject.Id);
+            Assert.IsNotNull(editedProject);
+
+            Assert.AreEqual(dbProject.Id, editedProject.Id);
+            SerializerAssert.AreNotEqual(dbProject, editedProject);
+        }
+
+        [Test]
+        public void SuccessfulEditProjectNoChangesTest()
+        {
+            existingProject.Id = Guid.NewGuid();
+
+            editProject.Id = existingProject.Id;
+            editProject.Name = existingProject.Name;
+            editProject.IsActive = existingProject.IsActive;
+            editProject.Description = existingProject.Description;
+            editProject.Deferred = existingProject.Deferred;
+            editProject.DepartmentId = existingProject.DepartmentId;
+
+            dbContext.Add(existingProject);
+            dbContext.SaveChanges();
+            dbContext.Entry(existingProject).State = EntityState.Detached;
+            dbContext.SaveChanges();
+
+            var dbProject = dbContext.Projects
```

Comment:
???

Reply:
I mentioned it on the call...

Two entities cannot be tracked at the same time so I detach the first one, that was added.